pkg/loki/loki.go

```diff
@@ -621,9 +627,11 @@ func (t *Loki) setupModuleManager() error {
 		TableManager:             {Server, Analytics},
 		Compactor:                {Server, Overrides, MemberlistKV, Analytics},
 		IndexGateway:             {Server, Store, Overrides, Analytics, MemberlistKV, IndexGatewayRing, IndexGatewayInterceptors},
+		BloomCompactor:           {Server, BloomCompactorRing, Analytics},
 		IngesterQuerier:          {Ring},
 		QuerySchedulerRing:       {Overrides, Server, MemberlistKV},
 		IndexGatewayRing:         {Overrides, Server, MemberlistKV},
+		BloomCompactorRing:       {Overrides, MemberlistKV},
```

Comment:
`IndexGatewayRing` adds the `Server` dependency despite the `IndexGateway` already having the  `Server` dependency. Should `BloomCompactorRing` also add `Server` as well?

Reply:
I see your point. But I can't justify why they were added in the first place? They already have `MemberlistKV` which has `Server` dependency. i prefer we remove it from the others as well, unless there's another reason i may be missing. wdyt?

Reply:
I think you should add everything you need just in case a transient dependency changes ? but anyway this is not thing that changes much often so 🤷 not important.